universal-application-tool-0.0.1/app/auth/SamlCiviFormProfileAdapter.java

```diff
@@ -0,0 +1,172 @@
+package auth;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSet;
+import java.util.ArrayList;
+import java.util.Locale;
+import java.util.Optional;
+import javax.inject.Provider;
+import models.Account;
+import models.Applicant;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.core.credentials.Credentials;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.core.profile.creator.AuthenticatorProfileCreator;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
+import org.pac4j.saml.client.SAML2Client;
+import org.pac4j.saml.config.SAML2Configuration;
+import org.pac4j.saml.profile.SAML2Profile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import repository.UserRepository;
+
+public class SamlCiviFormProfileAdapter extends AuthenticatorProfileCreator {
+
+  private static Logger LOG = LoggerFactory.getLogger(SamlCiviFormProfileAdapter.class);
+  protected final ProfileFactory profileFactory;
+  protected final Provider<UserRepository> applicantRepositoryProvider;
+  protected final SAML2Configuration saml2Configuration;
+  protected SAML2Client saml2Client;
+
+  public SamlCiviFormProfileAdapter(
+      SAML2Configuration configuration,
+      SAML2Client client,
+      ProfileFactory profileFactory,
+      Provider<UserRepository> applicantRepositoryProvider) {
+    super();
+    this.profileFactory = Preconditions.checkNotNull(profileFactory);
+    this.applicantRepositoryProvider = applicantRepositoryProvider;
+    this.saml2Client = client;
+    this.saml2Configuration = configuration;
+  }
+
+  @Override
+  public Optional<UserProfile> create(
+      Credentials cred, WebContext context, SessionStore sessionStore) {
+    ProfileUtils profileUtils = new ProfileUtils(sessionStore, profileFactory);
+
+    Optional<UserProfile> samlProfile = super.create(cred, context, sessionStore);
+
+    if (samlProfile.isEmpty()) {
+      LOG.warn("Didn't get a valid profile back from SAML");
+      return Optional.empty();
+    }
+
+    if (!(samlProfile.get() instanceof SAML2Profile)) {
+      LOG.warn(
+          "Got a profile from SAML2 callback but it wasn't a SAML profile: %s", samlProfile.get());
```

Comment:
the `.get()` returns a UserProfile (samlProfile is an Optional) and here we're asserting that it's a SAML2Profile instead of some other implementation of UserProfile. This code was mostly copied from `CiviformProfileAdapter.java`